collector/accelerators.go

```diff
@@ -142,3 +140,30 @@ func isMonitoredAccelerator(vendor, device, pciID string) (cardData, bool) {
 	}
 	return cardData{vendorData.vendorName, deviceDesc, pciID}, true
 }
+
+func prepareVendorModelData(configmapPath string) (map[string]vendorData, error) {
+	yamlStr, err := os.ReadFile(configmapPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to open accelerators config file %s: %v", configmapPath, err)
+	}
+	var vendorsModelsConfig []vendorModels
+	err = yaml.Unmarshal(yamlStr, &vendorsModelsConfig)
+	if err != nil {
+		return nil, fmt.Errorf("failed to unmarshal accelerators config data: %v", err)
+	}
+	vendorToDeviceMap := make(map[string]vendorData, len(vendorsModelsConfig))
+
+	for _, vendorModelsConfig := range vendorsModelsConfig {
```

Comment:
can we return an error if we detect duplicate vendor and PCI ids?

Reply:
Using `yaml.UnmarshalStrict` could be helpful in other cases as well.

Reply:
@machine424 which other cases are you referring to? We only have to unmarshal the configmap once. Or fo you mean to use UnmarshalStrict instread of Unmarshal?

Reply:
yes, instead of Unmarshal to help with incorrect fields names, malformed yaml etc. In addition to adding to the checks that Simon suggested.